miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/permission/MiniAppCustomPermissionCache.kt

```diff
@@ -80,22 +80,44 @@ internal class MiniAppCustomPermissionCache(context: Context) {
      * Stores the grant results to SharedPreferences.
      * @param [miniAppCustomPermission] an object to contain the results per MiniApp.
      */
-    @SuppressLint("RestrictedApi")
     fun storePermissions(
         miniAppCustomPermission: MiniAppCustomPermission
     ) {
         val supplied = miniAppCustomPermission.pairValues.toMutableList()
-
         // Remove any unknown permission parameter from HostApp.
         supplied.removeAll { (first) ->
             first.type == MiniAppCustomPermissionType.UNKNOWN.type
         }
-
         val miniAppId = miniAppCustomPermission.miniAppId
         val allPermissions = prepareAllPermissionsToStore(miniAppId, supplied)
         applyStoringPermissions(MiniAppCustomPermission(miniAppId, allPermissions))
     }
 
+    fun updateManifestPermissions(
+        miniAppId: String,
+        manifestPermissions: List<Pair<MiniAppCustomPermissionType, MiniAppCustomPermissionResult>>
+    ) {
```

Comment:
The function name makes it sound like this is updating the cached manifest - but it's actually updating the cached permissions. How about `updateFromManifestPermissions` instead?

Or actually - this class doesn't even need to know that these are manifest permissions. It just needs to know that it should remove permissions which weren't passed. So how about this?
```suggestion
    fun removePermissionsNotMatching(
        miniAppId: String,
        permissions: List<Pair<MiniAppCustomPermissionType, MiniAppCustomPermissionResult>>
    ) {
```